src/app/chat/view.nim

```diff
@@ -508,22 +510,60 @@ QtObject:
     echo self.status.mailservers.setMailserver("16Uiu2HAm4v86W3bmT1BiH6oSPzcsSr24iDQpSN5Qa992BCjjwgrD")
     echo self.status.mailservers.requestAllHistoricMessages()
 
+  proc switchTo*(self: ChatsView, communityId: string, channelId: string, 
+    messageId: string) =
+    ## This method displays community with communityId as an active one (if 
+    ## communityId is empty, "Chat" section will be active), then displays 
+    ## channel/chat with channelId as an active one and finally display message
+    ## with messageId as a central message in the message list.    
+    if (communityId.len > 0):
+      self.communities.setActiveCommunity(communityId)
+      if (channelId.len > 0):
+        self.channelView.setActiveChannel(channelId)
+        if (messageId.len > 0):
+          self.messageView.switchToMessage(messageId)
+    else:
+      self.communities.activeCommunity.setActive(false)
+      if (channelId.len > 0):
+        self.channelView.setActiveChannel(channelId)
+        if (messageId.len > 0):
+          self.messageView.switchToMessage(messageId)
```

Comment:
It seems that this could be done that way:
```
if (communityId.len > 0):
      self.communities.setActiveCommunity(communityId)
else:
      self.communities.activeCommunity.setActive(false)

if (channelId.len > 0):
     self.channelView.setActiveChannel(channelId)
     if (messageId.len > 0):
          self.messageView.switchToMessage(messageId)

To avoid repetition on channe view and message view

Reply:
@alaibe your comment is definitely in place, but this is done this way intentionally (also doing it either way no any performance benefit), cause this may be changed in future, and it's important for a dev to easily understand the intention. I wanted to explicitly emphasize two similar but very different sections, one is "Chat" section which we're handling in `else` block, and the other is "Community" section (refers to all added communities), which we're handling in `then` block.

Reply:
My comment, just prove the fact that it is not really standard and looks quite weird from a reading code point of view.
Not a big deal, as you said: no performance benefit